docs/proposals/2019-02-08 - Fallback Content Elements.md

```diff
@@ -0,0 +1,94 @@
+# Alternatives to Raw HTML
+
+# Problem
+
+The content element `raw_html` is often used to inject Javascript into an ANS document which makes it difficult for downstream applications to render natively. Even if the application tries to render it as a embeded browser, there is no guarentee that it will render correctly in all applications. And some applications (ex Apple TV or $50 Kindle) may not have the input or processing capabilities needed to effectively render the experience.
+
+# Proposal
+
+Add an `alternatives` trait to the `raw_html` content element, that contains one or more arrays of content elements that can notionally replace the `raw_html` block. The `basic` one is the only one required, and cannot contain a `raw_html` content element. `upvert`ing should fill this with a stock `text` element that has a stock `This content is not available on this platform` message (or similar).
+
+## Alternatives
+
+Each `raw_html` element should have an `alternatives` and at least a `basic` array with non-`raw_html` content. An example of the default behaviour:
+
+```
+{
+  "type": "raw_html",
+  "content": "<script src=\"https://www.washingtonpost.com/stat/cool.js\"></script>",
+  "alternatives": {
+    "basic": [
+      {
+        "type": "text",
+        "content": "<b>This content is unavailable on this platform</b>"
+      }
+    ]
+  }
+}
+```
+
+# Example 1
+
+The graphics team has created a JS rendered graphic that scales to browser viewport size. They also generate static `.png`s as a fallback for mobile and tablet.
+
+```
+{
+  "type": "raw_html",
+  "content": "<div class=\"graphics\"><!-- Graphics HTML and JS --></div>",
+  "alternatives": {
+    "basic": [
```

Comment:
what are the cases we will use other keys?
Since it's a fallback, will it be simpler we just have one alternative?

Reply:
The intention was to leave it open to other things like AR or some other platform specific content. Less so as a fallback and more as a alternative set of content if the platform supported it. I guess I was oscillating between the more radical opinion that `raw_html` as an alternative to pure ANS content, and wrote this such that you could do it either way. If this is too confusing, let me know and I'll update it to lean more one way than the other. Related to @gengel 's comment too.